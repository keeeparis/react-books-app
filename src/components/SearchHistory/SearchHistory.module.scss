.History {
  position: absolute;
  inset: 45px 0 0 0;
  transition: all 0.2s ease;
  opacity: 0;
  visibility: hidden;
  z-index: 1;

  &.visible {
    opacity: 1;
    visibility: visible;
  }
}

@media (max-width: 660px) {
  .History {
    top: 55px;
  }
}

.HistoryItem {
  display: flex;
  align-items: center;
  position: relative;
  background-color: white;
  cursor: pointer;
  transition: all 0.2s ease;
  padding: 8px 12px;

  &:hover {
    background-color: lightgray;
  }

  &:first-child {
    font-size: 0.8em;
    font-style: italic;
    cursor: default;
    border-top-left-radius: 8px;
    border-top-right-radius: 8px;

    &:hover {
      background-color: white;
    }
  }

  &:last-child {
    border-bottom-left-radius: 8px;
    border-bottom-right-radius: 8px;
  }
}

.RemoveButton {
  transform: scale(1.3);
  margin-right: 10px;
  color: black;
  width: 20px;
  height: 20px;
  transition: all 0.2s ease;

  &:hover {
    background-color: white;
    border-radius: 50%;
  }
}

@media (max-width: 660px) {
  .HistoryItem {
    padding: 12px 18px;
    font-size: 1.2em;
  }

  .RemoveButton {
    transform: scale(1.8);
  }
}
