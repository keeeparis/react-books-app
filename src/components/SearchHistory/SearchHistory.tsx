import React, { MouseEvent } from 'react'
import cn from 'classnames'

import styles from './SearchHistory.module.scss'
import { useTranslation } from 'react-i18next'

interface SearchHistoryProps {
  history: string[]
  handleHistorySearchClick: (option: string) => () => void
  deleteFromSearchHistory: (
    option: string
  ) => (e: MouseEvent<HTMLButtonElement>) => void
  historyVisible: boolean
}

const SearchHistory = ({
  history,
  handleHistorySearchClick,
  deleteFromSearchHistory,
  historyVisible,
}: SearchHistoryProps) => {
  const { t } = useTranslation()

  return (
    <div className={cn(styles.History, { [styles.visible]: historyVisible })}>
      <div className={styles.HistoryItem}>{t('search-history')}</div>

      {history.map((el) => (
        <div
          key={el}
          onClick={handleHistorySearchClick(el)}
          className={styles.HistoryItem}
        >
          <button
            onClick={deleteFromSearchHistory(el)}
            className={styles.RemoveButton}
            type="button"
          >
            &#215;
          </button>

          {el}
        </div>
      ))}
    </div>
  )
}

export default SearchHistory
