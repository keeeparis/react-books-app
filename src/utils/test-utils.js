import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
