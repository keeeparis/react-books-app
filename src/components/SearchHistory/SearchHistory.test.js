import { render, screen, fireEvent, userEvent } from 'test-utils'
import SearchHistory from './SearchHistory'

const props = {
  history: ['far', 'boo', 'baz'],
  handleHistorySearchClick: jest.fn(),
  deleteFromSearchHistory: jest.fn(),
  historyVisible: false,
}

it('test SearchHistory Component', () => {
  const { asFragment } = render(<SearchHistory {...props} />)

  expect(asFragment()).toMatchSnapshot()
})
