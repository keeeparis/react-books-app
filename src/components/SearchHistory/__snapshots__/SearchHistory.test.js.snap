// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearchHistory Component 1`] = `
<DocumentFragment>
  <div
    class="History"
  >
    <div
      class="HistoryItem"
    >
      Search history
    </div>
    <div
      class="HistoryItem"
    >
      <button
        class="RemoveButton"
        type="button"
      >
        ×
      </button>
      far
    </div>
    <div
      class="HistoryItem"
    >
      <button
        class="RemoveButton"
        type="button"
      >
        ×
      </button>
      boo
    </div>
    <div
      class="HistoryItem"
    >
      <button
        class="RemoveButton"
        type="button"
      >
        ×
      </button>
      baz
    </div>
  </div>
</DocumentFragment>
`;
